HHS/OPRE-OPS
# Services Components with api tie-in

## What changed

Refactors Services Components to work with api.

## Issue

#1967 

## How to test

1. Create a new Agreement
2. Add Services Components
3. See them persist in backend

## Screenshots

_If relevant, e.g. for a front-end feature_

## Definition of Done Checklist
